node/pkg/common/scissors_test.go

```diff
@@ -78,3 +91,101 @@ func TestSupervisor(t *testing.T) {
 		)
 	}
 }
+
+func TestRunWithScissorsCleanExit(t *testing.T) {
+	ctx := context.Background()
+	errC := make(chan error)
+
+	itRan := false
+	RunWithScissors(ctx, errC, "TestRunWithScissorsCleanExit", func(ctx context.Context) error {
+		itRan = true
+		return nil
+	})
+
+	time.Sleep(100 * time.Millisecond)
+	require.Equal(t, true, itRan)
```

Comment:
waiting on a channel would be cleaner than sleeping a fixed amount of time

Reply:
I actually thought about waiting on a channel rather than timing. The problem is that the channel could wake up as soon as the go routine posts to it, which could be before the scissor code completes, and in this case it is important to let the scissors code complete to confirm it doesn't hang.

Reply:
That makes sense. We should be good then.